helloworld/notebooks/OpHousingPrices.ipynb

```diff
@@ -0,0 +1,386 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import org.apache.spark.SparkConf\n",
+    "import org.apache.spark.sql.SparkSession\n",
+    "import org.apache.spark.SparkContext\n",
+    "import org.apache.spark.sql.functions.udf\n",
+    "\n",
+    "import com.salesforce.op._\n",
+    "import com.salesforce.op.features._\n",
+    "import com.salesforce.op.features.types._\n",
+    "import com.salesforce.op.evaluators.Evaluators"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import com.salesforce.op.OpWorkflow\n",
+    "import com.salesforce.op.evaluators.Evaluators\n",
+    "import com.salesforce.op.readers.DataReaders"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Instantiate a SparkSession"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val conf = new SparkConf().setMaster(\"local[*]\").setAppName(\"HousingPricesPrediction\")\n",
+    "implicit val spark = SparkSession.builder.config(conf).getOrCreate()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us create a case class to describe the schema for the data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "case class HousingPrices(\n",
+    "  lotFrontage: Double,\n",
+    "  area: Integer,\n",
+    "  lotShape: String,\n",
+    "  yrSold : Integer,\n",
+    "  saleType: String,\n",
+    "  saleCondition: String,\n",
+    "  salePrice: Double)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then define the set of raw features that we would like to extract from the data. The raw features are defined using FeatureBuilders, and are strongly typed. TransmogrifAI supports the following basic feature types: Text, Numeric, Vector, List , Set, Map. In addition it supports many specific feature types which extend these base types: Email extends Text; Integral, Real and Binary extend Numeric; Currency and Percentage extend Real. For a complete view of the types supported see the Type Hierarchy and Automatic Feature Engineering section in the Documentation.\n",
+    "\n",
+    "Basic FeatureBuilders will be created for you if you use the TransmogrifAI CLI to bootstrap your project as described here. However, it is often useful to edit this code to customize feature generation and take full advantage of the Feature types available (selecting the appropriate type will improve automatic feature engineering steps).\n",
+    "\n",
+    "When defining raw features, specify the extract logic to be applied to the raw data, and also annotate the features as either predictor or response variables via the FeatureBuilders.\n",
+    "\n",
+    "SalesType encoder from text to numeric."
```

Comment:
"SalesType encoder from text to numeric." What?

perhaps expand a bit as in the Iris notebook:

```
"#### Feature Engineering\n",
"\n",
"We then define the set of raw features that we would like to extract from the data. The raw features are defined using [FeatureBuilders](https://docs.transmogrif.ai/Developer-Guide#featurebuilders), and are strongly typed. TransmogrifAI supports the following basic feature types: `Text`, `Numeric`, `Vector`, `List` , `Set`, `Map`. \n",
"In addition it supports many specific feature types which extend these base types: Email extends Text; Integral, Real and Binary extend Numeric; Currency and Percentage extend Real. For a complete view of the types supported see the Type Hierarchy and Automatic Feature Engineering section in the Documentation.\n",
"\n",
"Basic `FeatureBuilders` will be created for you if you use the TransmogrifAI CLI to bootstrap your project as described here. However, it is often useful to edit this code to customize feature generation and take full advantage of the Feature types available (selecting the appropriate type will improve automatic feature engineering steps).\n",
"\n",
"When defining raw features, specify the extract logic to be applied to the raw data, and also annotate the features as either predictor or response variables via the FeatureBuilders:"```

Reply:
changed